### ISSUE TITLE
umlaut (e.g. ü) in name of directory causes problem after sharing the directory to group

### ISSUE THREAD

**Issue by fuchsstein** [nextcloud/server#12069]
### Steps to reproduce
1. Create an directory with an german umlaut as "ä" or "ü". In my case "UniLübeck" 
2. Upload a PDF-file into this directory or a subdirectory.
3. share the directory with the umlaut to a group.

### Expected behaviour
They should be able to open the document.

### Actual behaviour
 All members of the group I shared the directory with (besides me) cannot open the document while I (owner of the directory) can.
`PDF.js v1.9.426 (build: 2558a58d)Message: Unexpected server response (500)`  is shown to them.

### Server configuration
**Operating system**:
debian 9
**Web server:**
Apache/2.4.25 (Debian) 
**Database:**
10.1.26-MariaDB-0+deb9u1 Debian 9.1
**PHP version:**
PHP 7.3.
**Nextcloud version:**
14.0.3
**Updated from an older Nextcloud/ownCloud or fresh install:**
updated

If other details are needed please inform me.

**Comment by fuchsstein** [comment]
I cannot reproduce the error anymore; even by doing the same steps I recall doing the first time and using the precise same names.

By that I close this bug report because I failed collecting enough information the time it did not work – for which I am sorry.

(
To answer your question anyway:
One user uses Firefox Quantum 63.0 64-bit on Arch Linux 4.18.16 under which it did not work.
Under 60.2.0esr (64-bit) on Kali GNU/Linux Rolling (my system) it worked with the user sharing the directory and failed with a testing user I created for reproduction.
The error appeared with every Document I tested (random sample of ~10 out of ~50).
)
